### PR TITLE
Test/config

### PR DESCRIPTION
**Pull request by khailtq-0688**

### CODE REVIEW

File: configs/UIT_VFSC/sentiment/BiGRU/s1/config_vipherv2_BiGRU_UIT_VFSC_sentiment.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  dropout: 0.2
+  input_dim: 512
+  label_smoothing: 0.1
+  model_type: GRU
+  name: BiGRU_Model3layer_UIT_VFSC_vipher_sentiment
+  num_layer: 3
+  num_output: 3
+  embedder:
+    bidirectional: 2
+    dropout: 0.3
```
Comment (hieuhm2) [inline configs/UIT_VFSC/sentiment/BiGRU/s1/config_vipherv2_BiGRU_UIT_VFSC_sentiment.yaml]: tách embedder config ra độc lập với model

File: models/utils.py [hunk 2]
```diff
@@ -41,55 +41,56 @@ def forward(self, x):
...
+        super().__init__()
+        self.embed_dim = config.embedder.embed_dim
         self.model_type = config.model_type
         self.bidirectional = config.embedder.bidirectional
         self.dropout_prob = config.embedder.dropout
         self.num_layer = config.embedder.num_layer
         self.device = config.device
         self.pad_idx = vocab.get_pad_idx
-        self.total_token_dict = vocab.total_token_dict
+        self.total_token_dict = vocab.total_tokens_dict
```
Comment (hieuhm2) [inline models/utils.py]: thống nhất có 's' hoặc không có 's'

File: test.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# from configs.utils import get_config
+
```
Comment (hieuhm2) [inline test.py]: push file này lên chi?

File: vocabs/VipherV2.py [hunk 4]
```diff
@@ -455,7 +455,7 @@ def decode_sentence(self, encoded_sentence: torch.Tensor) -> str:
             str: The reconstructed sentence.
         """
         # If there's a batch dimension, pick the first item
-        if encoded_sentence.ndim == 5:
+        if encoded_sentence.ndim == 3:
```
Comment (hieuhm2) [inline vocabs/VipherV2.py:448]: chỗ này giải thích thêm đi, dễ nhầm

File: vocabs/VipherV2.py [hunk 5]
```diff
@@ -471,6 +471,9 @@ def decode_sentence(self, encoded_sentence: torch.Tensor) -> str:
         current_word = []
         i = 0
         while i < len(encoded_sentence):
```
Comment (hieuhm2) [inline vocabs/VipherV2.py]: đoạn debug này cân nhắc loại bỏ/logging thay vì print()

**Review by hieuhm2 — CHANGES_REQUESTED** [review]

File: vocabs/VipherV2.py [hunk 6]
```diff
@@ -471,6 +471,9 @@ def decode_sentence(self, encoded_sentence: torch.Tensor) -> str:
...
+            token_values = encoded_sentence[i].tolist()
+            if len(token_values) != 5:
+                print(f"Lỗi tại token {i}: {token_values}")
```
Comment (khailtq-0688) [inline vocabs/VipherV2.py]:
```suggestion
```

File: models/utils.py [hunk 7]
```diff
@@ -17,6 +17,15 @@ def generate_padding_mask(seq, pad_token_id=0):
     return seq == pad_token_id
 
 
+def new_generate_padding_mask(seq, pad_token_id=0):
+    if seq.ndim == 3 and seq.shape[-1] == 5:
```
Comment (hieuhm2) [inline models/utils.py:22]: không cần phải viết thêm một hàm gen padding mask khác đâu, tìm cách gộp vào hàm có sẵn đi

File: models/Transformer_VipherV2.py [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+        # )
+        # self.embedding_onset = nn.Embedding(
+        #     num_embeddings=self.total_token_dict['onset'],
+        #     embedding_dim=self.d_model,
+        #     padding_idx=self.pad_idx
+        # )
+        # Linear map
+        # self.linear_map = nn.Linear(3 * self.d_model, self.d_model)
+
+        self.word_embbder = ViWordEmbedder(config, vocab)
```
Comment (hieuhm2) [inline models/Transformer_VipherV2.py:27]: Chỗ thì embedding chỗ thì word_embedder. Thống nhất thành embedder hết nhé.

**Review by hieuhm2 — CHANGES_REQUESTED** [review]

**Review by hieuhm2 — APPROVED** [review]
